RNWCPP/Desktop/WebSocket.cpp

```diff
@@ -251,13 +251,13 @@ void BaseWebSocket<Protocol, Socket, Resolver>::Stop()
 template<typename Protocol, typename Socket, typename Resolver>
 void BaseWebSocket<Protocol, Socket, Resolver>::EnqueueWrite(const string& message, bool binary)
 {
+  post(m_context, [this, message = std::move(message), binary]()
```

Comment:
What is the lifetime of "this"? should it be captured in a shared_ptr or weak pointer?

Reply:
IWebSocket is exposed as `std::unique_ptr`, because its containing `facebook::xplat::module::CxxModule` is also expose as `unique_ptr`.

Thus, I don't think we can instance this class as `shared_ptr`.